utils/config/encryption.go

```diff
@@ -0,0 +1,221 @@
+	if err != nil {
+		return err
+	}
+	if !secFileExists {
+		return errorutils.CheckError(errors.New(decryptErrorPrefix + "security file was not found"))
+	}
+	if key == "" {
+		return errorutils.CheckError(errors.New(decryptErrorPrefix + "security file does not contain a master key"))
+	}
+	return handleSecrets(config, decrypt, key)
+}
+
+// Encrypt the config file if it is decrypted while security file exists and contains a master key.
+func updateEncryptionIfNeeded(originalConfig *ConfigV2) error {
+	masterKey, _, err := getMasterKeyFromSecurityFile()
+	if err != nil || masterKey == "" {
+		return err
+	}
+
+	// Marshalling and unmarshalling to get a new separate config struct, to prevent modifying the config for the rest of the execution.
+	decryptedContent, err := originalConfig.getContent()
+	if err != nil {
+		return err
+	}
+	tmpEncConfig := new(ConfigV2)
+	err = json.Unmarshal(decryptedContent, &tmpEncConfig)
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+	err = saveConfig(tmpEncConfig)
+	if err != nil {
+		return err
+	}
+	// Mark that config file is encrypted
+	originalConfig.Enc = true
+	return nil
+}
+
+// Encrypt/Decrypt all secrets in the provided config, with the provided master key.
+func handleSecrets(config *ConfigV2, handler secretHandler, key string) error {
+	var err error
+	for _, rtDetails := range config.Artifactory {
+		rtDetails.Password, err = handler(rtDetails.Password, key)
+		if err != nil {
+			return err
+		}
+		rtDetails.AccessToken, err = handler(rtDetails.AccessToken, key)
+		if err != nil {
+			return err
+		}
+		rtDetails.ApiKey, err = handler(rtDetails.ApiKey, key)
+		if err != nil {
+			return err
+		}
+		rtDetails.SshPassphrase, err = handler(rtDetails.SshPassphrase, key)
+		if err != nil {
+			return err
+		}
+		rtDetails.RefreshToken, err = handler(rtDetails.RefreshToken, key)
+		if err != nil {
+			return err
+		}
+	}
+	if config.Bintray != nil {
+		config.Bintray.Key, err = handler(config.Bintray.Key, key)
+		if err != nil {
+			return err
+		}
+	}
+	if config.MissionControl != nil {
+		config.MissionControl.AccessToken, err = handler(config.MissionControl.AccessToken, key)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func getMasterKeyFromSecurityFile() (key string, secFileExists bool, err error) {
+	secFile, err := cliutils.GetJfrogSecurityFilePath()
+	if err != nil {
+		return "", false, err
+	}
+	exists, err := fileutils.IsFileExists(secFile, false)
+	if err != nil || !exists {
+		return "", false, err
+	}
+
+	config := viper.New()
+	config.SetConfigType("yaml")
+	f, err := os.Open(secFile)
+	if err != nil {
+		return "", false, errorutils.CheckError(err)
+	}
+	err = config.ReadConfig(f)
+	if err != nil {
+		return "", false, errorutils.CheckError(err)
+	}
+	key = config.GetString(masterKeyField)
+	return key, true, nil
+}
+
+func readMasterKeyFromConsole() (string, error) {
+	print("Please enter your master key to obtain the config token: ")
```

Comment:
Please enter your master key to obtain the config token:
-->
Please enter the master key: